migrations/1_initial_migration.js

```diff
@@ -1,7 +1,5 @@
-/* globals artifacts */
-
-const Migrations = artifacts.require('./Migrations.sol');
-
 module.exports = function deploy(deployer) {
-  deployer.deploy(Migrations);
+  deployer.then(async () => {
+    // yeah we won't be using this...
```

Comment:
Why not?
Fyi you can use `truffle migrate -f{the step number to start from}` to run all migrations starting from a given step

Reply:
because it's completely useless unless you store the Migrations artifact so it can look up the address of the migration contract on the network. With our versioned registry, we've replaced Migrations with something that is much more powerful and which makes the Migrations contract obsolete.

This script basically makes it so that running `migrate` *always* runs from the beginning and puts the registry and all the contracts into the correct state.